dev-scripts/deploy_custom_rh-che.sh

```diff
@@ -0,0 +1,304 @@
+#!/bin/bash
+# Copyright (c) 2018 Red Hat, Inc.
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+
+usage="\e[93;1m$(basename "$0") \e[0;1m{-5} [-u <username>] [-p <passwd>] \e[90m{-t <tag>} {-r <registry>} {-n} {-s} {-h} \e[0m-- Login to dev cluster and deploy che with specific build tag
+
+\e[32;1mwhere:\e[0m
+    \e[1m-5\e[0m  \e[93muse che-server V5 \e[31;1m!!MUST COME AS FIRST FLAG!!\e[0m
+    \e[1m-u\e[0m  \e[93musername for openshift account\e[0m
+    \e[1m-p\e[0m  \e[93mpassword for openshift account\e[0m
+    \e[1m-h\e[0m  show this help text
+    \e[1m-n\e[0m  do not delete files after script finishes
+    \e[1m-s\e[0m  wipe sql database (postgres)
+    \e[1m-t\e[0m  [\e[1mdefault=latest\e[0m] tag for specific build (first 7 characters of commit hash)
+    \e[1m-r\e[0m  docker image registry from where to pull
+
+\e[32;1mrequirements\e[0m:
+    \e[1moc\e[0m  openshift-client
+    \e[1mjq\e[0m  json CLI processor (dnf install jq)
+    \e[1myq\e[0m  yaml CLI processor based on jq (pip install yq)"
+
+export RH_CHE_DEPLOY_SCRIPT_CLEANUP="true";
+export RH_CHE_WIPE_SQL="false";
+export RH_CHE_OPENSHIFT_URL=https://dev.rdu2c.fabric8.io:8443;
+export RH_CHE_IS_V_FIVE="false";
+export RH_CHE_JDBC_USERNAME=pgche;
+export RH_CHE_JDBC_PASSWORD=pgchepassword;
+export RH_CHE_JDBC_URL=jdbc:postgresql://postgres:5432/dbche;
+
+function setVars() {
+  if [ "$RH_CHE_IS_V_FIVE" == "true" ]; then
+    export RH_CHE_PROJECT_ID="c82f44b-fabric8-9cc154c";
+    export RH_CHE_DOCKER_IMAGE_REGISTRY="rhche/che-server-multiuser";
+    export RH_CHE_PROJECT_NAMESPACE=che5-automated;
+    export RH_CHE_GITHUB_BRANCH=master;
+    export ECLIPSE_CHE_GITHUB_BRANCH=5.22.x;
+  else
+#    export RH_CHE_PROJECT_ID="nightly-fabric8";
+#    export RH_CHE_DOCKER_IMAGE_REGISTRY="dfestal/che-server";
+    export RH_CHE_PROJECT_ID="67da9ee-fabric8-ce16ce1";
+    export RH_CHE_DOCKER_IMAGE_REGISTRY="rhche/rh-che-server";
+    export RH_CHE_PROJECT_NAMESPACE=che6-automated;
+    export RH_CHE_GITHUB_BRANCH=rh-che6;
+    export ECLIPSE_CHE_GITHUB_BRANCH=6.2.x;
+  fi
+}
+
+function unsetVars() {
+  unset RH_CHE_SERVER_USERNAME;
```

Comment:
I think the name of this variable is misleading. If I understand correctly this is **not** a Che server username but rather the OpenShift cluster username. And anyway you should provide the option to only provide a token instead of username/password.

Reply:
renamed.
Can you use origin-clients `oc` with a token?